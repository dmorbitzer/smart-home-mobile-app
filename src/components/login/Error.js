import { Text } from "@react-native-material/core";
import { StyleSheet, View } from "react-native";

export default function Error(props) {
  if (props.hasError) {
    return (
      <View style={styles.errorView}>
        <Text style={styles.errorText}>Username oder Passwort ung√ºltig!</Text>
      </View>
    );
  }

  return <View style={styles.errorPlaceholder} />;
}

const styles = StyleSheet.create({
  errorView: {
    marginLeft: 12,
    backgroundColor: "#f56653",
    height: 40,
    justifyContent: "center",
    borderRadius: 10,
  },
  errorText: {
    textAlign: "center",
  },
  errorPlaceholder: {
    height: 40,
  },
});
