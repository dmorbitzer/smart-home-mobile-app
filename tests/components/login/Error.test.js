import React from "react";
import renderer from "react-test-renderer";

import Error from "../../../src/components/login/Error";

jest.useFakeTimers();

test("renders Error with hasError correctly", () => {
  const tree = renderer.create(<Error hasError />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders Error without hasError correctly", () => {
  const tree = renderer.create(<Error />).toJSON();
  expect(tree).toMatchSnapshot();
});
