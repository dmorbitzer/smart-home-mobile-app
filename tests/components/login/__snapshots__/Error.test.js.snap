// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Error with hasError correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f56653",
      "borderRadius": 10,
      "height": 40,
      "justifyContent": "center",
      "marginLeft": 12,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "letterSpacing": 0.5,
        },
        {
          "color": "#000000",
        },
        {
          "textAlign": "center",
        },
      ]
    }
  >
    Username oder Passwort ung√ºltig!
  </Text>
</View>
`;

exports[`renders Error without hasError correctly 1`] = `
<View
  style={
    {
      "height": 40,
    }
  }
/>
`;
